server/shared/src/main/scala/org/http4s/server/middleware/DefaultHead.scala

```diff
@@ -35,17 +36,47 @@ import org.http4s.Method.HEAD
   * requiring more optimization should implement their own HEAD handler.
   */
 object DefaultHead {
-  def apply[F[_]: Functor, G[_]: Concurrent](http: Http[F, G])(implicit F: MonoidK[F]): Http[F, G] =
+  def apply[F[_]: Functor, G[_]: Applicative](
+      http: Http[F, G]
+  )(implicit F: MonoidK[F]): Http[F, G] =
     Kleisli { req =>
       req.method match {
         case HEAD => http(req) <+> http(req.withMethod(GET)).map(drainBody[G])
         case _ => http(req)
       }
     }
 
-  def httpRoutes[F[_]: Concurrent](httpRoutes: HttpRoutes[F]): HttpRoutes[F] =
+  @deprecated("Use overload with Applicative constraint", "0.23.17")
+  def apply[F[_], G[_]](
+      http: Http[F, G],
+      F: Functor[F],
+      G: Concurrent[G],
+      M: MonoidK[F],
+  ): Http[F, G] = {
+    implicit val functor = F
+    implicit val monoidK = M
+
+    Kleisli { req =>
+      req.method match {
+        case HEAD => http(req) <+> http(req.withMethod(GET)).map(r => drainBody[G](r)(G))
+        case _ => http(req)
+      }
+    }
```

Comment:
Can this just delegate?

```suggestion
    apply(http)(F, G, M)
```

Reply:
If this compiles, I like it 👍🏻